opm/input/eclipse/EclipseState/Grid/EclipseGrid.cpp

```diff
@@ -2015,6 +2021,45 @@ std::vector<double> EclipseGrid::createDVector(const std::array<int,3>& dims, st
     }
 
 
+    void EclipseGrid::init_children_host_cells_logical(){       
+        auto  IJK_location = [](const std::size_t&  nx, const std::size_t& ny,const std::size_t& nz, 
+                                          const std::size_t& host_nx, const std::size_t& host_ny, const std::size_t& host_nz,
+                                          const std::size_t& base_host_nx, const std::size_t& base_host_ny, const std::size_t& base_host_nz){
+            auto [i_list, j_list, k_list] = VectorUtil::generate_cartesian_product(0,nx-1,0, ny-1,0, nz-1);
+            std::function<std::size_t(std::size_t, std::size_t)> div = std::divides<std::size_t>{};
+            std::function<std::size_t(std::size_t, std::size_t)> sum = std::plus<std::size_t>{};
```

Comment:
Storing these in `std::function<>` objects introduces a bit of unnecessary overhead.  You could just pass
`std::divides<>{}` and `std::plus<>{}` directly as arguments to `vectorScalarOperation()` instead, i.e., as
```C++
auto resultI = VectorUtil::vectorScalarOperation(i_list, nx / host_nx, std::divides<std::size_t>{});
resultI = VectorUtil::vectorScalarOperation(resultI, base_host_nx, std::plus<std::size_t>{});
```